### ISSUE TITLE
-s switch issue once strict is enabled

### ISSUE THREAD

**Issue by atoomic** [atoomic/perl#196]
I noticed this issue while running `run/switches.t`

Consider this program, note the `-s` flag
```perl
# cat t.t
#!perl -s
our ($x,$y);
BEGIN { print $x,$y; exit }
```

When running it using alpha-dev-02-strict@5e5ff3e1 this is raising some strict issues when parsing the arguments
```
> ./perl -Ilib t.t -x=foo
Global symbol "$x" requires explicit package name (did you forget to declare "my $x"?) at t.t line 1.
BEGIN not safe after errors--compilation aborted at t.t line 3.
```

**Comment by atoomic** [comment]
backtrace output once breaking to `gv.c`
```
             qerror(Perl_mess(aTHX_
                  "Global symbol \"%s%" UTF8f
                  "\" requires explicit package name (did you forget to "
```

gdb output
```
(gdb) bt
#0  S_find_default_stash (stash=0x7fffffffd748,	name=0x7fffffffe344 "x=foo", len=1, is_utf8=0, add=1, sv_type=SVt_PV) at gv.c:1877
#1  0x0000000000460b90 in Perl_gv_fetchpvn_flags (nambeg=0x7fffffffe344 "x=foo", full_len=1, flags=1, sv_type=SVt_PV) at gv.c:2406
#2  0x0000000000455712 in Perl_init_argv_symbols (argc=1, argv=0x7fffffffdfd0) at perl.c:4539
#3  0x000000000047f347 in yyl_fake_eof (fake_eof=0, bof=true, s=0xa18bc2 "") at	toke.c:7186
#4  0x00000000004881f4 in yyl_try (s=0xa18bc2 "") at toke.c:8771
#5  0x000000000048a0c5 in Perl_yylex ()	at toke.c:9438
#6  0x000000000049673b in Perl_yyparse (gramtype=258) at perly.c:340
#7  0x0000000000450acb in S_parse_body (env=0x0, xsinit=0x41fabe <xs_init>) at perl.c:2581
#8  0x000000000044f878 in perl_parse (my_perl=0x9ee010,	xsinit=0x41fabe <xs_init>, argc=4, argv=0x7fffffffdfb8,	env=0x0) at perl.c:1872
#9  0x000000000041fa25 in main (argc=4,	argv=0x7fffffffdfb8, env=0x7fffffffdfe0) at perlmain.c:126
```

**Comment by atoomic** [comment]
This is happening in Perl_init_argv_symbols

**Comment by atoomic** [comment]
fixed by 1546f46068c0e3403be8b4d6474a5bf19f97b5f8
